rust/relay/server/src/server.rs

```diff
@@ -1038,6 +1047,17 @@ where
                     debug_assert_eq!(&existing_n, number, "internal state should be consistent");
                 }
 
+                // Also delete from eBPF maps when deleting channel due to allocation removal
+                if c.bound {
```

Comment:
Also caught this potential bug?

Reply:
Good catch! I reckon we don't need to make it conditional though. The channel was bound previously and re-emitting this event doesn't hurt.

The comment is a bit too specific as well. This code path here doesn't know about eBPF. I reckon just remove the comment, it is actually clear that we should emit this event here.